cocos/base/CCRef.cpp

```diff
@@ -111,7 +116,7 @@ void Ref::release()
     --_referenceCount;
 
 #if CC_ENABLE_SCRIPT_BINDING && CC_ENABLE_GC_FOR_NATIVE_OBJECTS
-    if (_scriptOwned && _referenceCount==1 && _rooted)
+    if (_scriptOwned && _rooted && _referenceCount==_referenceCountAtRootTime)
```

Comment:
You are right... We don't know the release order, but also we don't know if unrooting when rc==1 is the best solution since it can leak... in any case we need more testing.